apps/mobile/src/components/toast/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { useImperativeHandle, useState } from 'react';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, { useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated';
+
+import CheckmarkCircle from '@/assets/checkmark-circle.svg';
+import { ResponsiveValue, useTheme } from '@shopify/restyle';
+
+import { Box, Text, Theme } from '@leather.io/ui/native';
+
+import { ToastData, ToastMethods, ToastProps, ToastType } from './types';
+
+const AnimatedBox = Animated.createAnimatedComponent(Box);
+
+const TOP_POSITION = {
+  closed: -100,
+  opened: 100,
+};
+
+export function Toast({ toastRef }: ToastProps) {
+  const theme = useTheme<Theme>();
+  const [toastData, setToastData] = useState<ToastData | null>(null);
+  const top = useSharedValue<number>(TOP_POSITION.closed);
+
+  useImperativeHandle<ToastMethods, ToastMethods>(toastRef, () => ({
+    display(data) {
+      setToastData(data);
+      top.value = withTiming(TOP_POSITION.opened);
+    },
+  }));
+
+  function getIcon(type: ToastType) {
```

Comment:
I think we could move `getIcon` + `getBackground` out of the main `Toast` scope into their own standalone helpers